lib/expand/plan.go

```diff
@@ -105,10 +105,9 @@ func (p *Peer) getOperationPlan(ctx operationContext) (*storage.OperationPlan, e
 		builder.AddPostHookPhase(plan)
 	}
 
-	// if added a master node, make sure it participates in leader election
-	if builder.JoiningNode.IsMaster() {
-		builder.AddElectPhase(plan)
-	}
+	// Enabele/disable leader election depending on the cluster role
+	// of the joining node
+	builder.AddElectPhase(plan)
```

Comment:
If this is only done for master nodes, it will not be possible to rejoin a node with a different role. The election status should be reset regardless of cluster role.